integration/env/common.go

```diff
@@ -86,31 +96,30 @@ func init() {
 		panic("version bundle version  must not be empty")
 	}
 	os.Setenv(EnvVarVersionBundleVersion, VersionBundleVersion())
+
+	// ClusterID returns a cluster ID unique to a run integration test. It might
```

Comment:
@rossf7 I cherry picked the commit you made :) https://github.com/giantswarm/aws-operator/pull/1964/commits/dc9ad6a8965e53234eddd450232ab2e062be90a5

I've updated it now, PTAL

Reply:
Btw with this approach ClusterID is effectively initialized once (doesn't change) for the lifetime of a test, it's a singleton.

Current tests start single cluster or recreate same cluster.

Also it's expected by the current test logic which calls/retrives Cluster ID is that the same value gets returned with every call for the lifetime of a test.

There is also requirement that multiple aws-operator e2e tests even for same commit/branch can run in parallel without collision.

If I'm not mistaken length reduction requirement comes from AWS resource name constraints - we use cluster id as prefix in lots of places, to uniquely / meaningfully / consistently identify test cluster related AWS resources created from to a given test.

I think this approach (with randomization and singleton) fulfills all these requirements.

Reply:
Nice! Yes the ClusterID needs to stay the same for the lifetime of the test.

The shortening is because of the 32 char limit on AWS resources. See discussion here https://github.com/giantswarm/aws-operator/pull/1888/files#r326227819.